### PR TITLE
JS: handle case-insensitive lodash imports

### PR DESCRIPTION
**Pull request by asger-semmle**
NPM package names are always lower-case, so the package `lodash.mergewith` does not match the member `_.mergeWith`. This PR makes `LodashUnderscore::member` case insensitive.

### CODE REVIEW

**Comment by asger-semmle** [comment]
Adding `bindingset` to `LodashUnderscore::member` was technically a breaking change, which was also why the test failed. So instead of doing that, I've now listed the names of all lodash members in a predicate.

[Evaluation on big-apps](https://git.semmle.com/asger/dist-compare-reports/tree/Asgers-MacBook-Pro.local_1544659865312) looks ok.

PTAL

**Comment by asger-semmle** [comment]
The test failure seems to be from Jenkins having trouble, but there's no rush to land this so I'll just retry the tests after 1.19 is out.

**Comment by esben-semmle** [comment]
Time for a retry?

**Comment by asger-semmle** [comment]
Should be ready to land - the test failures are for C++ so I assume they're unrelated.

File: javascript/ql/src/semmle/javascript/frameworks/LodashUnderscore.qll [hunk 1]
```diff
@@ -38,6 +38,319 @@ module LodashUnderscore {
    * In addition, the global variable `_` is assumed to refer to `lodash` or `underscore`.
    */
   DataFlow::SourceNode member(string name) { result.(Member).getName() = name }
+
+  /**
+   * Holds if `name` is the name of a member exported from the `lodash` package
+   * which has a corresponding `lodash.xxx` NPM package.
+   */
+  private predicate isLodashMember(string name) {
```
Comment (esben-semmle) [inline javascript/ql/src/semmle/javascript/frameworks/LodashUnderscore.qll]: Could we document how to update this? Ideally as the union of all exports from all lodash versions.
Maybe that documentation belongs outside of this file.

Comment (asger-semmle) [reply javascript/ql/src/semmle/javascript/frameworks/LodashUnderscore.qll]: I've added a comment mentioning how it was generated.

**Pull request merged**